source/preface.tex

```diff
@@ -0,0 +1,184 @@
+
+This document was prepared by
+Joint Technical Committee ISO/IEC JTC 1, \textit{Information technology},
+Subcommittee SC 22, \textit{Programming languages, their environments and system software interfaces}.
+
+This sixth edition cancels and replaces the fifth edition
+(ISO/IEC 14882:2017), which has been technically revised.
+It also incorporates the Technical Specifications
+\begin{itemize}
+\item ISO/IEC TS 19217:2015 \textit{Programming Languages --- \Cpp{} Extensions for concepts},
+\item ISO/IEC TS 21425:2017 \textit{Programming Languages --- \Cpp{} Extensions for ranges},
+\item ISO/IEC TS 22277:2017 \textit{Programming Languages --- \Cpp{} Extensions for Coroutines},
+\item ISO/IEC TS 21544:2018 \textit{Programming Languages --- Extensions to \Cpp{} for Modules},
+\item portions of ISO/IEC TS 19571:2016 \textit{Programming Languages --- Technical specification for \Cpp{} extensions for concurrency}, and % latch and barrier
+\item portions of ISO/IEC TS 19568:2017 \textit{Programming Languages --- \Cpp{} Extensions for Library Fundamentals}.
+\end{itemize}
+
+The main changes compared to the previous edition are as follows:
+
+\begin{itemize}
+% language changes
+\item addition of concepts, \grammarterm{requires-clause}s, and \grammarterm{requires-expression}s and
+\libheaderref{concepts} header
+\item addition of coroutines, including
+\keyword{co_yield}, \keyword{co_await}, and \keyword{co_return} keywords and
+\libheaderref{coroutine} header
+\item addition of modules, \grammarterm{import-declaration}s, and \grammarterm{export-declaration}s
+\item addition of three-way comparison, defaulted comparisons,
+rewriting of comparison operator expressions, and \libheaderref{compare} header
+\item addition of designated initializers
+\item support for class types and floating-point types as the type of a non-type template parameter
+\item new attributes \tcode{[[no_unique_address]]}, \tcode{[[likely]]}, \tcode{[[unlikely]]}
+\item support for optional reason string in \tcode{[[nodiscard]]} attribute
+\item ability to require constant initialization with \keyword{constinit} keyword
+\item ability to require constant evaluation with \keyword{consteval} keyword
+\item extension of constant evaluation support to cover
+\begin{itemize}
+  \item memory allocation and deallocation
+  \item virtual function calls
+  \item \keyword{dynamic_cast} and \keyword{typeid}
+  \item changing the active member of a union
+  \item uninitialized variables
+\end{itemize}
+\item support for constexpr functions to contain
+\keyword{catch} handlers and \grammarterm{asm-declaration}s
+that are not reached during constant evaluation
+\item support for controlling destruction in a class-specific operator delete function
+\item addition of \keyword{using} \keyword{enum} declaration
+\item addition of \keyword{char8_t} type
+\item guarantee that \keyword{char16_t} and \keyword{char32_t} literals
+are encoded as UTF-16 and UTF-32 respectively
```

Comment:
Should there be a comma before "respectively"?

Reply:
Added.